core/chains/evm/evm_txm.go

```diff
@@ -0,0 +1,101 @@
+package evm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/smartcontractkit/sqlx"
+
+	txmgrtypes "github.com/smartcontractkit/chainlink/common/txmgr/types"
+	evmclient "github.com/smartcontractkit/chainlink/core/chains/evm/client"
+	evmconfig "github.com/smartcontractkit/chainlink/core/chains/evm/config"
+	httypes "github.com/smartcontractkit/chainlink/core/chains/evm/headtracker/types"
+	"github.com/smartcontractkit/chainlink/core/chains/evm/logpoller"
+	"github.com/smartcontractkit/chainlink/core/chains/evm/txmgr"
+	evmtypes "github.com/smartcontractkit/chainlink/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/core/logger"
+)
+
+// TxManagerEvmType is embedded inside evmTxm, which requires use of this type aliasing.
+type TxManagerEvmType = txmgr.TxManager[*evmtypes.Head]
+
+var _ httypes.HeadTrackable = &evmTxm{}
+
+// evmTxm is an evm wrapper over the generic TxManager interface
+type evmTxm struct {
+	httypes.HeadTrackable
```

Comment:
let's use `txmgrtypes.Head[*evmtypes.Head]` here instead of `httypes.HeadTracakable`. (just tried locally and it works) so new structs use generic interfaces

Reply:
Actually, `txmgrtypes.TxManager` already requires `txmgrtypes.HeadTrackable[HEAD]` so I think `httypes.HeadTracakable` is redundant?

Reply:
Nvm, disregard above. I see what you're trying to accomplish by making the evmTxm compatible with the eth-specific code

Reply:
Any reasons we don't just change functions that expect `httypes.HeadTrackable` to expect `txmgrtypes.Head[*evmtypes.Head]`?